src/storage/s3.rs

```diff
@@ -105,15 +105,18 @@ impl Backend {
             prefix.pop();
         }
 
-        // `Region::default` will get try to get the region from the environment
-        // and fallback to a default if it isn't found.
-        let mut region = Region::default();
-        if let Ok(endpoint) = std::env::var("AWS_S3_ENDPOINT") {
-            region = Region::Custom {
-                name: region.name().to_owned(),
+        // If a custom endpoint is set, do not use the AWS default (us-east-1)
+        // instead, check environment variables for a region name.
+        let region = if let Ok(endpoint) = std::env::var("AWS_S3_ENDPOINT") {
+            Region::Custom {
+                name: std::env::var("AWS_DEFAULT_REGION")
+                    .or_else(|_| std::env::var("AWS_REGION"))
+                    .unwrap(),
```

Comment:
Is there anything we can do about this `.unwrap()`? If neither `AWS_DEFAULT_REGION` nor `AWS_REGION` are set, shouldn't it fall back to `us-east-1`? Something like this perhaps?
```rust
std::env::var("AWS_DEFAULT_REGION")
    .or_else(|_| std::env::var("AWS_REGION"))
    .unwrap_or_else(|| String::from("us-east-1")),
```

Reply:
I had a feeling you'd pick on that!

So I was going to do a default but it felt like it makes no sense to default to `us-east-1` when that region isn't going to exist on a custom S3 service. For example, Scaleway uses regions named `fr-par` and `nl-ams`, Digital Ocean uses `nyc3`, `fra1` etc. So it felt better to throw an error here.

Now, I must admit I couldn't figure out how to actually _return_ a Future that yields an error - I'm super rusty with the Futures API!